### PR TITLE
feat: add chain config

### PR DESCRIPTION
**Pull request by DoubleOTheven**
Resolves #

- [x] There is an associated issue (**required**) [Wallet Compatibility](https://github.com/paritytech/useink/issues/47)
- [x] The change is described in detail
- [ ] There are new or updated tests validating the change (if applicable)

## Description

Add new chain configurations in `useink/chains`. These configurations were modified from [chaindata](https://github.com/TalismanSociety/chaindata) so that we can have strongly typed chain keys. e.g. `useBlockHeader('acala')`.

There are about 300 chains added. Half of them are testnets and are separated by exports. We also export `PalletContractChains` and `PalletContractTestnets` for ease of use with chains that support PalletContracts in Substrate.

-------------------

## Examples

* [Config](https://github.com/paritytech/useink-kitchen-sink/blob/3380df8c91ae7067043f838940c7dcaf83c7752a/frontend/src/pages/_app.tsx#L6)
* Now you can set a specific RPC for a specific chain [using this](https://github.com/paritytech/useink-kitchen-sink/blob/3380df8c91ae7067043f838940c7dcaf83c7752a/frontend/src/components/pg-home/HomePage.tsx#L34).

### CODE REVIEW

File: chains/types.ts [hunk 1]
```diff
@@ -1,27 +1,337 @@
...
+  | 'turing'
+  | 'uniarts'
+  | 'unique'
+  | 'unorthodox'
+  | 'westlake'
+  | 'xx'
+  | 'zeitgeist'
+  | 'zero';
+
+export type TestChainId =
```
Comment (peetzweg) [inline chains/types.ts:141]: This type can potentially be inferred dynamically via the using the `as const` typescript syntax to make the hardcoded network values readonly.

Here is an example I created: [TypeScript Playground](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBAYwgOwM7wOrHcZATAFWxmWHgF44BvAKDjgEs8AuOAcgHdjc8BaGYqRhsANHTjIAhgFtgrNlhz5R4yQiQBXZDHkAqAEwBWQwEYAnCvqoNAI1QJJyAKpQANvIAWMGGFTMA9P5cSngAdNZ2DsihDBD+logekgxoYMAIAIpQLu7sXj5+gbIwkniSJaFgklAMMACeMXEAjlBB3PgA+qhpCKiVyADmCa7lxACyZKXlklk5zOL0bPm+Af7FUxVVNfWN-i1tIR3rZSUdIwLooZJggwlQYL2sANoL7ByohQcC+Lz3CJUQVwAa1KEBgjQSi3en2C3zC1wYoRQADMUpJXLUGrF-GBbBiELx3pC3h9Vn9KikgcgIHhgLtYTx-KAwK5oMAoMTOKTAgyfuS8FxXBioKEkNJOdDVry+H9+NEBcAhQwRWKJdz-PywQIoH0hBxoECvjw1Z9yQwbGBQnqDUblGJ6ABdMQAXzgklQiBQ6AA3DQaMzoPAkGh4AAlCBIJBEdBCOCUWj0JjyKAR1P8QRkBJSWTycORiAJNSabR6IymCz2uAReyOOZ5bwrQIp-PhWw16LYu4PPxwJ5c02ppC-B4A4Gg8GxNhOmiu92ekO+v3B9BwaMkMge+N51NroQiRRw3dkZ00eppOAAOTI+qgQIvMmAcbgQOAdQgyLgZ+A79XGZgqF9L9f3QABhJIUgASTwJ8vx-I9-yeK8YBvO8HwdPsmCnGggA)

[Article explaining usage of `as const`](https://medium.com/lean-coders/understand-const-as-const-and-readonly-in-typescript-a-comprehensive-explanation-1b54f24b35d6)

File: chains/types.ts [hunk 2]
```diff
@@ -1,27 +1,337 @@
...
+
+export type Account = '*25519' | 'secp256k1' | 'Sr25519';
+
+export type JsonString = string;
+
+export interface Token {
+  symbol: string;
+  decimals: number;
+  ed: string;
+  onChainId: JsonString | number;
```
Comment (peetzweg) [inline chains/types.ts:308]: Similar to `ed`, this is not self explanatory to me. 
We need to start with the docs anyways, we could right away already adding some JSDoc annotations to the codebase.

https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html

File: chains/types.ts [hunk 3]
```diff
@@ -1,27 +1,337 @@
...
-  testnet?: boolean;
-  coingeckoId?: string;
-  token: {
-    symbol: string;
+  account: Account;
+  subscanUrl?: string;
+  overrideNativeTokenId?: string;
+  chainspecQrUrl?: string;
+  latestMetadataQrUrl?: string;
+  rpcs: string[];
```
Comment (peetzweg) [inline chains/types.ts:326]: We could potentially narrow the `rpc` type even more using template literal types. 

https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html

This way we could enforce only valid rpc urls which are using `ws/wss` websockets. Does useink work with https rpcs?

File: chains/types.ts (hunk 1 above)
Comment (DoubleOTheven) [reply chains/types.ts:141]: This is much better 👍🏼

File: chains/types.ts [hunk 4]
```diff
@@ -1,27 +1,337 @@
...
-  | 'Phala'
-  | 'PhalaTestnet'
-  | 'Rococo'
-  | 'Shibuya'
-  | 'Shiden';
-
-export interface IChain {
+export type ChainName = string;
+
+export type ChainId = ProductionChainId | TestChainId;
```
[the remark this answers is not in the document]
Comment (DoubleOTheven) [reply chains/types.ts]: The `development` ChainId can be used for customized chain configs. 

I think that what we gain from type safety here is worth keeping for the sake of the majority of devs who are building Dapps, not the < 1% who are building substrate chains.

File: chains/types.ts (hunk 2 above)
Comment (DoubleOTheven) [reply chains/types.ts:23]: it is the `existentialDeposit`. I renamed it and added a comment

File: chains/types.ts (hunk 3 above)
Comment (DoubleOTheven) [reply chains/types.ts]: Done!

**Pull request merged**